strax/storage/mongo.py

```diff
@@ -11,100 +11,159 @@
+        """
         if MongoClient is None:
             raise ImportError("Pymongo did not import: "
                               "cannot use MongoDBStore")
         self.client = MongoClient(uri)
         self.db = self.client[database]
+        self.col_name = col_n
 
     def _read_chunk(self, backend_key, chunk_info, dtype, compressor):
+        """See strax.Backend"""
+        query = backend_key_to_query(backend_key)
+        chunk_name = f'chunk_{chunk_info["chunk_i"]}'
+        # Query for the chunk and project the chunk info
+        doc = self.db[self.col_name].find_one(
+            {**query, chunk_name: {"$exists": True}},
+            {f"{chunk_name}.$": 1})
+        doc = doc.get(chunk_name, None)
 
-        doc = self.db[backend_key].find_one(
-            {'chunk': chunk_info['chunk_i']})
         if doc is None:
-            return []
+            return np.array([], dtype=dtype)
 
-        result = np.zeros(len(doc['data']), dtype=dtype)
-        for i, d in enumerate(doc['data']):
+        # Convert JSON to numpy
+        result = np.zeros(len(doc), dtype=dtype)
+        for i, d in enumerate(doc):
             for key in np.dtype(dtype).names:
-                result[i][key] = d[key]
+                result[i][key] = d['data'][key]
         return result
 
     def _saver(self, key, metadata):
-        return MongoSaver(key, metadata, self.db[str(key)])
+        """See strax.Backend"""
+        # Use the key to make a collection otherwise, use the backend-key
+        col = self.db[self.col_name if self.col_name is not None else str(key)]
+        return MongoSaver(key, metadata, col)
 
     def get_metadata(self, key):
-        return self.db[key].find_one({'metadata': True})
+        """See strax.Backend"""
+        query = backend_key_to_query(key)
+        doc = self.db[self.col_name].find_one(query)
+        if doc and 'metadata' in doc:
+            return doc['metadata']
+        raise strax.DataNotAvailable
 
 
 @export
 class MongoFrontend(StorageFrontend):
     """MongoDB storage frontend"""
 
-    def __init__(self, uri, database, *args, **kwargs):
+    def __init__(self, uri, database, col_n=None, *args, **kwargs):
+        """
+        MongoFrontend for reading/writing data from Mongo
+        :param uri: Mongo url (with pw and username)
+        :param database: name of database (str)
+        :param col_n: collection name (str) to look for data
+        :param args: init for StorageFrontend
+        :param kwargs: init for StorageFrontend
+        """
+
         super().__init__(*args, **kwargs)
         if MongoClient is None:
             raise ImportError("Pymongo did not import: "
                               "cannot use MongoDBStore")
         self.client = MongoClient(uri)
         self.db = self.client[database]
-        self.backends = [MongoBackend(uri, database)]
+        self.backends = [MongoBackend(uri, database, col_n=col_n)]
+        self.col_name = col_n
```

Comment:
What's the argument for using `col_n` in the argument and `col_name` as the attribute? Wouldn't things be a bit simpler if they were the same?

Reply:
If you think its easier I see little reason not to change it so will do that. There was a reason why I did it like this but I cannot remember.

Reply:
I don't have a strong opinion on this, I'm just thinking about readability.